sbi/samplers/vi/vi_sampling.py

```diff
@@ -0,0 +1,198 @@
+import torch
+from torch import Tensor
+from torch.distributions import Distribution
+import numpy as np
+
+from typing import Optional, List, Callable
+
+import warnings
+
+from .vi_quality_controll import gpdfit, GerneralizedParetto
+
+
+_SAMPLING_METHOD = {}
+_SAMPLING_PARAMETERS_DOC = {}
+
+
+def register_sampling_method(
+    cls: Optional[Callable] = None,
+    name: Optional[str] = None,
+    doc: Optional[str] = "",
+):
+    """Registers a sampling method which can be used to debias the variational posterior.
+
+    Args:
+        cls: The method to add.
+        name: The name of the method
+        doc: A short description, which will be shown to the user.
+
+
+    """
+
+    def _register(cls: Callable):
+        if name is None:
+            cls_name = cls.__name__
+        else:
+            cls_name = name
+        if cls_name in _SAMPLING_METHOD:
+            raise ValueError(f"The sampling method {cls_name} is already registered")
+        else:
+            _SAMPLING_METHOD[cls_name] = cls
+            _SAMPLING_PARAMETERS_DOC[cls_name] = doc
+        return cls
+
+    if cls is None:
+        return _register
+    else:
+        return _register(cls)
+
+
+def get_sampling_method(name: str) -> Callable:
+    """Returns the sampling_function of the specified method
+
+    Args:
+        name: The name of the sampling method.
+
+    Returns:
+        Callable: Returns the sampling method.
+
+    """
+    return _SAMPLING_METHOD[name]
+
+
+def get_sampling_method_parameters_doc(name: str) -> str:
+    """Returns a description of a sampling method method
+
+    Args:
+        name: The name of the sampling method.
+
+    Returns:
+        str: Short description of the method.
+
+    """
+    return _SAMPLING_PARAMETERS_DOC[name]
+
+
+def get_default_sampling_methods() -> List[str]:
+    """Returns the list of all registered sampling methods!
+
+    Returns:
+        List[str]: Registered sampling methods.
+
+    """
+    return list(_SAMPLING_METHOD.keys())
+
+
+def paretto_smoothed_weights(weights: Tensor) -> Tensor:
```

Comment:
Why is this in `sampling`? Imo this and `clamp_weights()` are very different from `sir` and `naive`, so I would have them in different files

Reply:
I'd move them to vi_utils

Reply:
Yeah, I just placed them there because its importance *sampling* :D

Reply:
hm okey :D I think that's a bit confusing. We should have a dedicated module for samplers that are "like SIR"

Reply:
Mhh, we can make something like vi_is.py and put there gdpfit, Paretto distribution and all these kind of stuff.

Reply:
I guess if we remove the weight_f support form fKLOptimizer then we also can completly remove these.